### PR TITLE
MG-93 - Add support for creating charts

### LINKED ISSUE #93: Add support for creating charts

### PR DESCRIPTION
**Pull request by Musilah**
# What type of PR is this?

- [ ] Refactor
- [X] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Dependency Update

## What does this do?
This creates chart templates that will be added to the widgets. 

- Add Apache E-Charts to the Magistrala dashboards.
- The charts should be able to use the SENML data format used in Magistrala.

## Which issue(s) does this PR fix/relate to?

- Related Issue #81
- Resolves #93

## Have you included tests for your changes?

- [ ] Yes
- [X] No

## Did you document any new/modified functionality?

- [ ] Yes
- [X] No

### Notes
The charts being added include: 
1. Time series line chart
2. Time series bar graph
3. Analogue gauge
4. Radial temperature gauge
5. Pie chart
6. Donut chart
7. Dynamic line and bar chart
8. Stacked area line chart
9. Range chart
10. Analogue speed gauge
image

### CODE REVIEW

**Comment by dborovcanin** [comment]
@Musilah Please resolve the conflicts and mark as _Ready for review_ when it's ready.

**Review by dborovcanin — COMMENTED** [review]
Add a separate `templates` subdirectory for charts and charts modals.

File: package-lock.json [hunk 1]
```diff
@@ -5,15 +5,15 @@
   "packages": {
     "": {
       "devDependencies": {
-        "prettier": "^3.1.0",
```
Comment (ianmuchyri) [inline package-lock.json:8]: why are you removing the prettier dependency? It's used for prettier formatting

File: ui/jsonObject.go [hunk 2]
```diff
@@ -21,59 +21,65 @@ func CreateItem() []Item {
 			Widget:  "lineChart",
```
Comment (ianmuchyri) [inline ui/jsonObject.go:21]: rename this file to a more descriptive name.

Comment (ianmuchyri) [reply ui/jsonObject.go:21]: and remove the to be deleted comment

File: ui/jsonObject.go [hunk 3]
```diff
@@ -21,59 +21,65 @@ func CreateItem() []Item {
...
 		{
-			Title:   "Gauge",
+			Title:   "Temperature Gauge",
 			Content: "This is a small sentence about a line chart",
 			Image:   "https://example.com/image9.jpg",
-			Widget:  "gauge",
+			Widget:  "tempGauge",
 		},
 		{
 			Title:   "Line Chart",
```
Comment (ianmuchyri) [inline ui/jsonObject.go:72]: Line chart is being repeated a couple of times

**Review by ianmuchyri — CHANGES_REQUESTED** [review]

File: ui/charts.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+	Image   string `json:"image"`
+	Widget  string `json:"widget"`
+}
+
+func CreateItem() []Item {
+	items := []Item{
+		{
+			Title:   "Time Series Line Chart",
+			Content: "This is a small sentence about a line chart",
+			Image:   "https://example.com/image1.jpg",
```
Comment (ianmuchyri) [inline ui/charts.go:19]: remove the image field if we are not using it. If we are using it, find relevant open source images to use for the images

**Review by ianmuchyri — CHANGES_REQUESTED** [review]

File: ui/web/templates/charts/dynamicdatachartmodal.html [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+        form.reset();
+
+        var widgetID = "dynamicDatachart-" + Date.now();
+        var widgetScript = `
+        var dynamicDataChart = echarts.init(document.getElementById("${widgetID}"), null, {
+                width: 600,
+                height: 600,
+                });
+                var app = {};
+                const categories = (function () {
```
Comment (ianmuchyri) [inline ui/web/templates/charts/dynamicdatachartmodal.html:108]: use `var` instead of `const`

**Review by ianmuchyri — CHANGES_REQUESTED** [review]

**Review by dborovcanin — CHANGES_REQUESTED** [review]
Newly added widgets are not resizable. However, if I first save and after that enter edit mode, I can see the resize option.

**Review by ianmuchyri — APPROVED** [review]

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**